### PR TITLE
Display Lux Warning when Lux cannot create visualizations for empty or small dataframes

### PR DESCRIPTION
**Pull request by jinimukh**
This PR implements the frontend UI changes for displaying warnings for empty or small dataframes/series. Screenshots are attached below:

## Changed
- added new CSS style ids for the warning icon and box.
- display the message sent from backend if length of recommendations is 0. In this case, the message is always set to be the reason as to why no recommendations exist. In this case, it is because we are passing in empty/small dataframes.

Screen Shot 2021-02-26 at 1 09 39 PM
Screen Shot 2021-02-26 at 1 09 55 PM

### CODE REVIEW

File: src/warningBtn.tsx [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import React, { Component } from 'react';
 import { WarningBtnProps } from './interfaces/warningBtnProps';
+// import { constants } from 'zlib';
```
Comment (dorisjlee) [inline src/warningBtn.tsx:3]: remove comment

File: src/widget.tsx [hunk 2]
```diff
@@ -285,25 +285,26 @@ export class LuxWidgetView extends DOMWidgetView {
...
                                      setIntent={this.setIntent}
                                      closeExportInfo={this.closeExportInfo}
                                      tabItems={this.state.tabItems}
                                      showAlert={this.state.showAlert}
                                      intentEnabled={intentEnabled}
-                                     />               
-                </div>);
+                                     />
+                      <div id="no-recs-footer" style={{display:"flex"}}>
+                        <div id="no-recs" className = "fa fa-exclamation-triangle"></div>
```
Comment (dorisjlee) [inline src/widget.tsx:304]: Do we want to remove the bullet for the list item here? In this case, it is only possible to have a single bullet item right?

image

File: src/widget.tsx [hunk 3]
```diff
@@ -285,25 +285,26 @@ export class LuxWidgetView extends DOMWidgetView {
...
         if (this.state.recommendations.length == 0) {
           return (<div id="oneViewWidgetContainer" style={{ flexDirection: 'column' }}>
-                  {/* {attributeShelf}
-                  {filterShelf} */}
-                  <div style={{ display: 'flex', flexDirection: 'row' }}>
-                    <CurrentVisComponent intent={this.state.intent} currentVisSpec={this.state.currentVis} numRecommendations={0}
-                    onChange={this.handleCurrentVisSelect}/>
-                  </div>
-                  <ButtonsBroker buttonsEnabled={buttonsEnabled}
+                    <div style={{ display: 'flex', flexDirection: 'row' }}>
```
Comment (dorisjlee) [inline src/widget.tsx:290]: Is there a clean and easy way to make the warning box not span the full height of the CurrentVisComponent in this case? There is a lot of empty white space below the warning message box.

**Review by dorisjlee — CHANGES_REQUESTED** [review]

**Comment by dorisjlee** [comment]
One very strange issue that I found when printing out a bunch of `Vis` objects in series is that the no-rec-footer div shows up sometimes. We should probably hide this div unless it is necessary.
```python
df = pd.read_csv("../lux-datasets/data/airbnb_nyc.csv")
from lux.vis.Vis import Vis
for attr in ["room_type","neighbourhood_group","neighbourhood","host_name"]:
    for cattr in ["room_type","neighbourhood_group","neighbourhood","host_name"]:
        if attr!=cattr:
            vis = Vis([lux.Clause(attr,channel="y"), "number_of_reviews",lux.Clause(cattr,channel="color")], df)
            vis._repr_html_()
```
image

File: src/widget.tsx (hunk 3 above)
Comment (jinimukh) [reply src/widget.tsx:300]: This should be fixed in the latest push! @dorisjlee

File: src/widget.tsx (hunk 2 above)
Comment (jinimukh) [reply src/widget.tsx]: Fixed in next push!

**Comment by dorisjlee** [comment]
Thanks @jinimukh !

**Pull request merged**